lib/src/theme.dart

```diff
@@ -1,29 +1,105 @@
 part of flame_splash_screen;
 
-WidgetBuilder _whiteLogoBuilder = (context) => Image(
-      width: 300,
-      image: const AssetImage(
-        'assets/flame-logo-white.gif',
-        package: 'flame_splash_screen',
-      ),
+class AnimatedLogo extends AnimatedWidget {
+  const AnimatedLogo({
+    Key? key,
+    required Animation<double> animation,
+  }) : super(
+          key: key,
+          listenable: animation,
+        );
+
+  @override
+  Widget build(BuildContext context) {
+    final animation = listenable as Animation<double>;
+    return Stack(
+      fit: StackFit.expand,
+      alignment: Alignment.center,
+      children: [
+        Image.asset(
+          'assets/layer1.png',
+          package: 'flame_splash_screen',
+        ),
+        Opacity(
+          opacity: animation.value,
+          child: Image.asset(
+            'assets/layer2.png',
+            package: 'flame_splash_screen',
+          ),
+        ),
+        Image.asset(
+          'assets/layer3.png',
+          package: 'flame_splash_screen',
+        ),
+      ],
+    );
+  }
+}
+
+class LogoComposite extends StatefulWidget {
+  const LogoComposite({Key? key}) : super(key: key);
+
+  @override
+  _LogoCompositeState createState() => _LogoCompositeState();
+}
+
+class _LogoCompositeState extends State<LogoComposite>
+    with SingleTickerProviderStateMixin {
+  late Animation<double> animation;
+  late AnimationController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+      duration: const Duration(milliseconds: 500),
+      vsync: this,
     );
+    animation = Tween<double>(begin: 0.0, end: 1.0).animate(controller);
+
+    controller.addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        controller.reverse();
+      } else if (status == AnimationStatus.dismissed) {
+        controller.forward();
+      }
+    });
+
+    controller.forward();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
 
-WidgetBuilder _darkLogoBuilder = (context) => Image(
-      width: 300,
-      image: const AssetImage(
-        'assets/flame-logo-black.gif',
-        package: 'flame_splash_screen',
+  @override
+  Widget build(BuildContext context) => AnimatedLogo(
+        animation: animation,
+      );
```

Comment:
This method should have a body since it covers more than one line.

Reply:
Fixed